scent.py

```diff
@@ -0,0 +1,50 @@
+"""Configuration file for sniffer."""
+
+import time
+import subprocess
+
+from sniffer.api import select_runnable, file_validator, runnable
+
+try:
+    from pync import Notifier
+except ImportError:
+    notify = None
+else:
+    notify = Notifier.notify
+
+watch_paths = ['.']
+
+
+@select_runnable('python')
+@file_validator
+def py_files(filename):
+    return "TemplateDemo" not in filename
+
+
+@runnable
+def python(*_):
+    group = int(time.time())  # unique per run
+
+    for count, (command, title) in enumerate((
+            (('dmypy', 'run', "--", "xonsh"), "Lint"),
+            (('flake8', '--count'), "Lint"),
+            (('pytest', 'tests/test_main.py'), "Test main"),
+            (('pytest', 'tests/test_ptk_highlight.py'), "Test ptk highlight"),
+            (('pytest', '--ignore', 'tests/test_main.py', 'tests/test_ptk_highlight.py'), "Test Rest"),
+    ), start=1):
```

Comment:
Rather than running pytest, should it just run `run-tests.xsh`?

Reply:
it is a continuous test runner. Running all would take time. I used to hide/show each commands.

Reply:
I am not sure I understand. Can you explain a bit more please?

Reply:
https://pypi.org/project/sniffer/ 

it is the config file for this tool. 

it runs the commands on terminal as the code changes. so it is easier to refactor the large number of errors from mypy. It is a developer tool. Not going to end-up in the test-suit or the build tools .

Reply:
I guess my question is why are you running pytest directly rather than through the run-tests script? I would like to keep the list of test files that need to be run in a separate process in a single location

Reply:
run-tests also has flake8 and mypy 

it is not really a make file, where we selectively run each task.

Reply:
Ahh I see ok.